### PR TITLE
Messaging - service + controller

### PR DESCRIPTION
**Pull request by kaysoky**
See https://sagebionetworks.jira.com/wiki/display/PLFM/Messaging+services

### CODE REVIEW

**Comment by kaysoky** [comment]
Oops, still need to add tests.

File: services/repository/src/main/java/org/sagebionetworks/repo/web/controller/MessageController.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+/**
+ * <p>
+ * Provides REST APIs for sending messages to other Synapse users and for commenting on Synapse entities.
+ * </p>
+ */
+@ControllerInfo(displayName = "Message Services", path = "repo/v1")
+@Controller
+public class MessageController extends BaseController {
```
Comment (brucehoff) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/controller/MessageController.java:39]: I think there is an annotation you can put here (deprecated?) to suppress the auto generated doc's until done.  (delete the 'controllerinfo' annotation?)

File: services/repository/src/test/java/org/sagebionetworks/repo/web/controller/MessageControllerAutowiredTest.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+		status.setRecipientId(bobId);
+		status.setStatus(MessageStatusType.ARCHIVED);
+		ServletTestHelper.updateMessageStatus(bob, status);
+		
+		inboxOfBob = ServletTestHelper.getInbox(bob, inboxFilter, SORT_ORDER, DESCENDING, LIMIT, OFFSET);
+		assertEquals(0L, inboxOfBob.getTotalNumberOfResults());
+		assertEquals(0, inboxOfBob.getResults().size());
+		
+		// Eve can't interfere (does nothing)
+		status.setStatus(MessageStatusType.UNREAD);
```
Comment (brucehoff) [inline services/repository/src/test/java/org/sagebionetworks/repo/web/controller/MessageControllerAutowiredTest.java:249]: could you make the controller throw 403 Forbidden if the user changing the status is not a recipient? consider 404 if the message doesn't exist or hasn't yet been received.

File: integration-test/src/test/java/org/sagebionetworks/IT501SynapseJavaClientMessagingTest.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	}
+
+	@Test
+	public void testGetInbox() throws Exception {
+		PaginatedResults<MessageBundle> messages = synapseOne.getInbox(null,
+				null, null, LIMIT, OFFSET);
+		assertEquals(1, messages.getResults().size());
+		assertEquals(twoToOne, messages.getResults().get(0).getMessage());
+
+		messages = synapseTwo.getInbox(null, null, null, LIMIT, OFFSET);
```
Comment (brucehoff) [inline integration-test/src/test/java/org/sagebionetworks/IT501SynapseJavaClientMessagingTest.java:106]: please fill in the null parameters to exercise passing the request param's

**Pull request merged**